narwhal/worker/src/quorum_waiter.rs

```diff
@@ -98,6 +104,7 @@ impl QuorumWaiter {
                     let (primary_names, worker_names): (Vec<_>, _) = workers.into_iter().unzip();
                     let message  = WorkerBatchMessage{batch: batch.clone()};
                     let handlers = self.network.broadcast(worker_names, &message);
+                    let timer = self.metrics.batch_broadcast_quorum_latency.start_timer();
```

Comment:
Probably won't make much of a difference, but I would suggest to put this before the `network.broadcast` command so on the unlikely event that broadcast it self also deals with some latency it can be captured on the spent time.

Reply:
I think we can measure `broadcast()` call latency explicitly if it turns out to be needed. I don't want to introduce too many variables into the metric.